backend/models/courseunit.mgmodel.ts

```diff
@@ -0,0 +1,105 @@
+import mongoose, { Schema, Document } from "mongoose";
+
+type ElementSkeleton = {
+  id: string;
+  x: number;
+  y: number;
+  w: number;
+  h: number;
+  content: string;
+};
+
+const ElementSkeletonSchema: Schema = new Schema({
+  x: {
+    type: Number,
+    required: true,
+  },
+  y: {
+    type: Number,
+    required: true,
+  },
+  w: {
+    type: Number,
+    required: true,
+  },
+  h: {
+    type: Number,
+    required: true,
+  },
+  content: {
+    type: String,
+    required: true,
+  },
+});
+
+type Page = {
+  id: string;
+  title: string;
+  type: "Lesson" | "Activity";
+  layout: [ElementSkeleton];
+};
+
+const PageSchema: Schema = new Schema({
+  title: {
+    type: String,
+    required: true,
+  },
+  type: {
```

Comment:
To match the `Page.type` type definition above, we'll want the type validator `String` and an enum validator `["Lesson", "Activity"]` (similar to what we do for `User.role`)